x/staking/client/cli/tx.go

```diff
@@ -25,7 +25,7 @@ var (
 	defaultCommissionRate          = "0.1"
 	defaultCommissionMaxRate       = "0.2"
 	defaultCommissionMaxChangeRate = "0.01"
-	defaultMinSelfDelegation       = "1"
+	defaultMinSelfDelegation       = "1" + sdk.DefaultBondDenom
```

Comment:
This is going to make `stake` the default flag which I don't think we should do.

Reply:
Yeah. Make sense.
But, I've refer to the `defaultAmount` which contains `sdk.DefaultBondDenom` as below:
https://github.com/youngjoon-lee/cosmos-sdk/blob/af995cd5ee5c30dccbb989cd6892e142d5dde818/x/staking/client/cli/tx.go#L24

Because I made a denom required, I think the `defaultMinSelfDelegation` has to have a denom.
Of course, I made `parseMinSelfDelegation()` to check whether the denom is the same as the expected one, or not.
What do you think about this strategy?

Reply:
Also, I thought this is a uncompleted improvement, as long as we change `MsgCreateValidator` to contains `MinSelfDelegation` as `sdk.Coin` (not `sdk.Int`): https://github.com/youngjoon-lee/cosmos-sdk/blob/09f9a5573e4132fbedfaab6d948df1cc93e6e896/x/staking/types/tx.pb.go#L43

While I wrote this PR, I've felt that the original author chose `sdk.Int` rather than `sdk.Coin` because he wanted to define `MinSelfDelegation` as "the minimum value regardless of what its denom is".
If so, I think it's okay to close this PR without merging.

Reply:
Yeah honestly, I don't think having a denom in the CLI UX will add that much benefit to be honest. I'm not sure what others think.

Reply:
Yeah. As the Gaia document has been updated: https://github.com/cosmos/gaia/pull/867, I think it's clear that the unit of `--min-self-delegation` is `stake` or `uatom`.

So, I'm closing this PR. Please feel free to open this PR if you think it's necessary.